### PR TITLE
Android Studio 3.5.0

### PR DESCRIPTION
**Pull request by hannesa2**
Simply bring library up to date with

* most recent stable Android Studio
* most recent stable Kotlin
* update libraries
* fix resulting compile errors

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -1,17 +1,11 @@
...
     <option name="LINE_SEPARATOR" value="&#10;" />
-    <AndroidXmlCodeStyleSettings>
-      <option name="USE_CUSTOM_SETTINGS" value="true" />
-    </AndroidXmlCodeStyleSettings>
     <JavaCodeStyleSettings>
       <option name="FIELD_NAME_PREFIX" value="m" />
       <option name="STATIC_FIELD_NAME_PREFIX" value="s" />
       <option name="CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND" value="99999" />
       <option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="99999" />
-      <option name="PACKAGES_TO_USE_IMPORT_ON_DEMAND">
```
Comment (davigonz) [inline .idea/codeStyles/Project.xml]: Why removing this?

File: .idea/codeStyles/Project.xml [hunk 2]
```diff
@@ -42,8 +36,6 @@
     <XML>
       <option name="XML_ATTRIBUTE_WRAP" value="0" />
       <option name="XML_KEEP_BLANK_LINES" value="1" />
-      <option name="XML_ALIGN_ATTRIBUTES" value="false" />
```
Comment (davigonz) [inline .idea/codeStyles/Project.xml]: And these options?

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (hannesa2) [reply .idea/codeStyles/Project.xml]: `3.5` changed the codestyle settings. This is automatic done by ide

File: .idea/codeStyles/Project.xml (hunk 2 above)
Comment (hannesa2) [reply .idea/codeStyles/Project.xml]: `3.5` changed the codestyle settings. This is automatic done by ide

**Comment by davigonz** [comment]
@hannesa2 the code looks good to me, please rebase it and we will merge it

**Comment by hannesa2** [comment]
> @hannesa2 the code looks good to me, please rebase it and we will merge it

done

**Pull request merged**